src/main/java/backend/IssueMetadata.java

```diff
@@ -3,27 +3,40 @@
 import github.TurboIssueEvent;
 import org.eclipse.egit.github.core.Comment;
 
+import util.Utility;
 import java.time.LocalDateTime;
 import java.time.ZoneOffset;
 import java.util.ArrayList;
+import java.util.Date;
 import java.util.List;
 
 public class IssueMetadata {
     private final List<TurboIssueEvent> events;
     private final List<Comment> comments;
     private final LocalDateTime nonSelfUpdatedAt;
+    private final LocalDateTime selfUpdatedAt;
     private final int nonSelfCommentCount;
+    private final int selfCommentCount;
 
-    // If isUpdated is true, nonSelfUpdatedAt will be used to sort/filter instead of updatedAt in TurboIssue
-    private final boolean isUpdated;
+    private final boolean isUpdated; //for any update
+    private final boolean isUpdatedBySelf; // for update by self
+    private final boolean isUpdatedByOthers; // for update by others
```

Comment:
Sorry, this again... is `isUpdated` equal to `isUpdatedBySelf || isUpdatedByOthers`? Like, is there a case where the values of these can differ? If not, we can get rid of this field and implement `isUpdated()` in terms of the other ones.